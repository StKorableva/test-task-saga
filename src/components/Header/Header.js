import React  from "react";
import "./Header.scss";
import CityLink from "../CityLink/CityLink";

const Header = () => {
  return (
    <header className="header">
      <CityLink
        city="/saint-petersburg"
        isActive={(match, location) => match || location.pathname === "/"}
      >
        Saint-Petersburg
      </CityLink>
      <CityLink city="/moscow">Moscow</CityLink>
      <CityLink city="/rostov-on-don">Rostov-On-Don</CityLink>
    </header>
  );
};

export default Header;
