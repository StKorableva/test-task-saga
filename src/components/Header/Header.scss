a {
  text-decoration: none;
}

.header {
  align-items: center;
  display: flex;
  height: 6vh;
  justify-content: center;
  list-style: none;
  width: 100%;
  z-index: 1;

  &-item {
    color: #000;
    font-family: "Arial", serif;
    font-size: 15px;
    letter-spacing: 1px;
    margin: 0 0px;
    padding-right: 15px;
    text-transform: uppercase;
  }
}
