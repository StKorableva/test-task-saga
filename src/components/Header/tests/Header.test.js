import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from '../Header';


const header = shallow(<Header />);

describe('<Header />', () => {
	it('should match snapshot', () => {
		expect(toJson(header)).toMatchSnapshot();
	});
});
