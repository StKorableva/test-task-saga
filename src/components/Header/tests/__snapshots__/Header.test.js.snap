// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<header
  className="header"
>
  <nav
    className="header__nav"
  >
    <ul
      className="header__nav-list"
    >
      <li
        className="header__nav-item"
      >
        <NavLink
          activeStyle={
            Object {
              "color": "red",
            }
          }
          isActive={[Function]}
          label="Saint-Petersburg"
          to="/saint-petersburg"
        >
          Saint-Petersburg
        </NavLink>
      </li>
      <li
        className="header__nav-item"
      >
        <NavLink
          activeStyle={
            Object {
              "color": "red",
            }
          }
          label="Moscow"
          to="/moscow"
        >
          Moscow
        </NavLink>
      </li>
      <li
        className="header__nav-item"
      >
        <NavLink
          activeStyle={
            Object {
              "color": "red",
            }
          }
          label="Rostov-on-Don"
          to="/rostov-on-don"
        >
          Rostov-on-Don
        </NavLink>
      </li>
    </ul>
  </nav>
</header>
`;
